StormDBManager.py

```diff
@@ -0,0 +1,296 @@
+            return ret
+
+        return self.db_lock.run(_execute, self, query, arguments, get_lastrowid)
+
+    def executemany(self, query, list):
+        """
+        Executes a query on the twisted thread-pool using the storm framework many times using the values provided by
+        a list.
+
+        Args:
+            query: The sql query to be executed.
+            list: The list containing tuples of values to execute the query with.
+
+        Returns: A deferred that fires once the execution is done, the result will be None.
+
+        """
+        def _execute(connection, query, arguments=None):
+            connection.execute(query, arguments, noresult=True)
+
+        @transact
+        def _executemany(self, query, list):
+            connection = Connection(self._database)
+            for item in list:
+                _execute(connection, query, item)
+            connection.close()
+        return self.db_lock.run(_executemany, self, query, list)
+
+    def executescript(self, sql_statements):
+        """
+        Executes a script of several sql queries sequentially.
+        Note that this function does exist in SQLite, but not in the storm framework:
+        https://www.mail-archive.com/storm@lists.canonical.com/msg00569.html
+
+        Args:
+            sql_statements: A list of sql statements to be executed.
+
+        Returns: A deferred that fires with None once all statements have been executed.
+
+        """
+        def _execute(connection, query):
+            connection.execute(query, noresult=True)
+
+        @transact
+        def _executescript(self, sql_statements):
+            connection = Connection(self._database)
+            for sql_statement in sql_statements:
+                _execute(connection, sql_statement)
+            connection.close()
+        return self.db_lock.run(_executescript, self, sql_statements)
+
+    def fetchone(self, query, arguments=None):
+        """
+        Executes a query on the twisted thread-pool using the storm framework and returns the first result.
+
+        Args:
+            query: The sql query to be executed.
+            arguments: Optional arguments that go with the sql query.
+
+        Returns: A deferred that fires with the first tuple that matches the query or None.
+        The result would be the same as using execute and calling the next() function on it, instead now you will get
+        None instead of a StopIterationException.
+
+        """
+        @transact
+        def _fetchone(self, query, arguments=None):
+            connection = Connection(self._database)
+            result = connection.execute(query, arguments).get_one()
+            connection.close()
+            return result
+
+        return self.db_lock.run(_fetchone, self, query, arguments)
+
+    def fetchall(self, query, arguments=None):
+        """
+        Executes a query on the twisted thread-pool using the storm framework and returns all results as a list of
```

Comment:
I would say that the deferred is called with the results which is a list of tuples.